types/posthtml.d.ts

```diff
@@ -28,23 +27,22 @@ export interface PostHTMLOptions {
   skipParse?: boolean;
 }
 
-type Plugin<T> = (tree: PostHTMLTree) => void | PostHTMLTree | ThisType<T>;
+export type PostHTMLPlugin<T> = (tree: PostHTMLNode) => void | PostHTMLNode | ThisType<T>;
 
-interface Result<M> {
+export interface Result<M> {
```

Comment:
💄 The `M` type-parameter also looks like non-speaking name. I prefer `TMessage` instead.